sphinx_needs/functions/functions.py

```diff
@@ -152,10 +163,11 @@ def find_and_replace_node_content(
     if found, check if it contains a function string and run/replace it.
 
     :param node: Node to analyse
-    :return: None
     """
     new_children = []
-    if (
+    if isinstance(node, NeedFunc):
+        return node.get_text(env, need)
+    elif (
         not node.children
         and isinstance(node, nodes.Text)
         or isinstance(node, nodes.reference)
```

Comment:
Below replace statements should contain a duplicate:
```
            func_string = func_string.replace("„", '"')
            func_string = func_string.replace("“", '"')
            func_string = func_string.replace("”", '"')  <- duplicates
            func_string = func_string.replace("”", '"')  <- duplicates
            func_string = func_string.replace("‘", "'")  # noqa: RUF001
            func_string = func_string.replace("’", "'")  # noqa: RUF001
```
Copied from raw file view to VS Code looks like this
![image](https://github.com/user-attachments/assets/6c9d872d-4ad6-4424-8705-2c9135d4467a)

I also cross checked this beast with a hex editor, they are identical. Maybe a different closing quote was meant.

Reply:
this is all required due to having to "undo" https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-smartquotes already run on the `Text`.
Yet another reason to get rid of the `[[...]]` 😅

Reply:
I would fix this in another PR though